Jenkinsfile

```diff
@@ -45,3 +46,18 @@ node {
     throw e
   }
 }
+
+// Run schema tests outside of 'node' definition, so that they do not block the
+// original executor while the downstream tests are being run
+if (env.BRANCH_NAME != 'master') {
```

Comment:
> This looks like the app tests are only run when not building master, which I think is different from how the CI works currently. Is this an intentional change?

Good spot! That's a mistake. I didn't realise that the app tests were triggered by the master build when I originally wrote that check last week - I should have spotted it when I updated it today. I've fixed it now.

> I think the overall job status is only dependent on all of the other builds succeeding when building the master branch.

This change is intentional: in the Jenkins 1 world, every app build reported its status to GitHub individually when triggered by a push to schemas. The Jenkins 2 pipeline builds aren't configured to individually update GitHub, so an app build failure needs to cause the parent schema build to fail, regardless of which branch is being built, so that the failure is visible on the PR.